aten/src/ATen/native/cuda/UnaryOpsKernel.cu

```diff
@@ -173,6 +173,35 @@ void clamp_max_kernel_cuda(TensorIterator& iter, Scalar max_value) {
   });
 }
 
+void heaviside_tensor_kernel_cuda(TensorIterator& iter) {
+  AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "heaviside_tensor_cuda", [&]() {
+    gpu_kernel(iter, [=]GPU_LAMBDA(scalar_t self, scalar_t val) -> float {
+       if (self > 0) {
```

Comment:
Maybe an inner loop like:

```
self == 0 ? val : (self > 0)
```

would be less code and possibly faster, since comparing to zero is generally faster than other comparisons.

Reply:
Addressed, Thanks for advice!